### ISSUE TITLE
Using ContentDialog in code without parented will make it not be keyboard accessible

### ISSUE THREAD

**Issue by madmrmax** [MicrosoftDocs/windows-uwp#2051]
All the examples with ContentDialog do not support keyboard dismissal and is not accessible. You need to have the ContentDialog parented to ensure it changes keyboard focus correctly.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 3361e67d-b904-0d10-42d7-abf317fbf0ab
* Version Independent ID: 993d923a-b11a-2f4e-6751-a0a266109e8b
* Content: [Dialog controls - Windows UWP applications](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/dialogs-and-flyouts/dialogs#feedback)
* Content Source: [windows-apps-src/design/controls-and-patterns/dialogs-and-flyouts/dialogs.md](https://github.com/MicrosoftDocs/windows-uwp/blob/docs/windows-apps-src/design/controls-and-patterns/dialogs-and-flyouts/dialogs.md)
* Product: **uwp**
* Technology: **design**
* GitHub Login: @jwmsft
* Microsoft Alias: **jimwalk**

**Comment by jevansaks** [comment]
This sounds like a product bug if true. Do you have a repro app that demonstrates this?

**Comment by madmrmax** [comment]
I found that it only repros when I call it from code within a PointerPressed event. I was just trying to do some "alert" styled check to figure out if my code was working, and saw that ContentDialog is the more recent way to do an alert. If I do the same code within the Tapped event it is fine.

        private async void dwtToolBox_PointerPressed(object sender, PointerRoutedEventArgs e)
        {
           // _toolboxItemInPress = Convert.ToInt32(Math.Floor(e.GetPosition(dwtToolBox).X / 58));

            ContentDialog cd = new ContentDialog
            {
                Content = "test", //_toolboxItemInPress.ToString(),
                CloseButtonText = "Close",
                DefaultButton = ContentDialogButton.Close
            };

            await cd.ShowAsync();
        }

**Comment by jevansaks** [comment]
Our interaction guidelines specify to act on pointer *up*, but even better would be to use the Click event (which covers all  input types) and that should work for mouse, touch, keyboard, etc.

**Comment by madmrmax** [comment]
> Our interaction guidelines specify to act on pointer _up_, but even better would be to use the Click event (which covers all input types) and that should work for mouse, touch, keyboard, etc.

Thanks. Image didn't have the Click event so I couldn't do that. In addition have you read the complexity in the docs of actually ensuring that your pointer released matches your pointer pressed? It seemed like overkill just to verify my calculations were accurate. I guess I could try to see if the MessageDialog UI works vs. the ContentDialog in the pressed event (again just for debugging).

**Comment by jevansaks** [comment]
> Thanks. Image didn't have the Click event so I couldn't do that. In addition have you read the complexity in the docs of actually ensuring that your pointer released matches your pointer pressed? It seemed like overkill just to verify my calculations were accurate.

Right, this is why we recommend using Button. :)

You could use Button (to get border/background) or ButtonBase (no visuals) and put an Image as the content.

**Comment by madmrmax** [comment]
> 
> You could use Button (to get border/background) or ButtonBase (no visuals) and put an Image as the content.

Thanks. It doesn't look like ButtonBase can be used in XAML and Button has too many pre-defined visuals.  Too bad there isn't like a Click behavior (almost like a CSS style) I could apply to Image to be able to use it.

I appreciate all your comments though!

**Comment by jwmsft** [comment]
If you need a button without all the visuals, you can use the built-in TextBlockButtonStyle ThemeResource.
```
<Button Style="{ThemeResource TextBlockButtonStyle}" Click="Button_Click">
    <Image Source="ms-appx:///Assets/StoreLogo.png"/>
</Button>
```
Although it's designed for text, it works with other button content.
